model/CSVDialog.cc

```diff
@@ -149,14 +148,13 @@ std::string CSVDialog::loadWebFile(const std::string& url)
   if (found!=std::string::npos) {
       vector<unsigned char> zbuf(res.get().body().size());
```

Comment:
This zbuf will be too small. The point of compressing data is to make it smaller, and you are sizing the receive buffer to be the compressed size. You should use the zs.total_out value computed after zs.deflate.

Reply:
Yes, I tried to do that in the lines following it, but I may have to assign a CHUNK of memory exceeding res.get.body().size() as discussed at https://www.zlib.net/zlib_how.html and then correct for it using zs.total_out, if I understand correctly:


            have = CHUNK - strm.avail_out;
            if (fwrite(out, 1, have, dest) != have || ferror(dest)) {
                (void)deflateEnd(&strm);
                return Z_ERRNO;
            }

Reply:
Yes - you have to do that crap when dealing with the C API. The point of my ZStream class is to handle that detail for you. It automatically resizes the buffer as it reads in the compressed data.